### PR TITLE
Codegen/IDL 4: definitions for a `Points2D` archetype

### PR DESCRIPTION
**Pull request by teh-cmc**
Nothing but definitions!

And I say "a" `Points2D` rather than "the" `Points2D` because it's not the point of this PR to nail down the perfect `Points2D` archetype (yet); but rather get close enough and stress-test the whole system in the process.

---

Codegen/IDL PR series:
- #2362
- #2363
- #2369
- #2370 
- #2374 
- #2375 
- #2410
- #2432

### CODE REVIEW

File: crates/re_types/definitions/rerun/archetypes/points2d.fbs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+include "rerun/datatypes.fbs";
+include "rerun/components.fbs";
+
+namespace rerun.archetypes;
+
+// ---
+
+// TODO(cmc): archetype IDL definitions must always be tables
+// TODO(cmc): archetype IDL definitions must refer to objects of kind component
```
Comment (teh-cmc) [inline crates/re_types/definitions/rerun/archetypes/points2d.fbs:11]: issues

Comment (teh-cmc) [reply crates/re_types/definitions/rerun/archetypes/points2d.fbs:11]: also `rerun.attr.component_required` should need `required`...

File: crates/re_types/definitions/rerun/attributes.fbs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+attribute "rerun.attr.component_required";
+
+/// Marks a component as recommended, which is likely to impact the generated code in
+/// backend-specific ways.
+///
+/// Only applies to the fields of an archetype.
+attribute "rerun.attr.component_recommended";
+
+/// Marks a component as optional, which is likely to impact the generated code in
+/// ways.
```
Comment (Wumpf) [inline crates/re_types/definitions/rerun/attributes.fbs:16]:
```suggestion
/// backend-specific ways.
```
unless the optional one is more mysterious than the others :D

**Review by Wumpf — APPROVED** [review]
neat. As recommended I focused on gauging how fbs works out here rather than the point specifics and I think this is all good to go!

as discussed on slack, let's put `attr` first on the attribute namespacing scheme

File: crates/re_types/definitions/rerun/components/draw_order.fbs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+include "python/attributes.fbs";
+include "rust/attributes.fbs";
+
+include "rerun/datatypes.fbs";
+include "rerun/attributes.fbs";
+
+namespace rerun.components;
+
+// ---
+
```
[the remark this answers is not in the document]
Comment (teh-cmc) [reply crates/re_types/definitions/rerun/components/draw_order.fbs:11]: Ahead of you on this one :sunglasses::
- https://github.com/rerun-io/rerun/issues/2388

**Comment by teh-cmc** [comment]
-> https://github.com/rerun-io/rerun/pull/2370/commits/af2482b18153021de9a7cb384fd90cf78c6be676

**Pull request merged**